src/client/Microsoft.Identity.Client/Http/Headers/AuthenticationHeaderParser.cs

```diff
@@ -0,0 +1,154 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Identity.Client.PlatformsCommon.Factories;
+using Microsoft.Identity.Client.Utils;
+
+namespace Microsoft.Identity.Client
+{
+    /// <summary>
+    /// Parsed authentication headers to retreve header values from HttpResponseHeaders.
+    /// </summary>
+    public class AuthenticationHeaderParser
```

Comment:
I think you should place this class in the root, where the `WWWAuthenticateParameteers` used to live.